File: src/main/java/com/snowflake/examples/de/Step05_FahrenheitToCelsiusUDF.java
package com.snowflake.examples.de;

public class Step05_FahrenheitToCelsiusUDF {

    /**
     * Converts Fahrenheit to Celsius.
     *
     * @param fahrenheit Temperature in Fahrenheit
     * @return Temperature in Celsius
     */
    public static double fahrenheitToCelsius(double fahrenheit) {
        return (fahrenheit - 32) * 5 / 9;
    }

    public static void main(String[] args) {
        double fahrenheit = 98.6;
        double celsius = fahrenheitToCelsius(fahrenheit);

        System.out.printf("%.2f°F is equal to %.2f°C%n", fahrenheit, celsius);
    }
}

File: src/main/scala/com/snowflake/examples/de/Step05_FahrenheitToCelsiusUDF.scala
package com.snowflake.examples.de

class Step05_FahrenheitToCelsiusUDF {

  /** Converts Fahrenheit to Celsius */
  def fahrenheitToCelsius(fahrenheit: Double): Double = (fahrenheit - 32) * 5 / 9

  def main(args: Array[String]): Unit = {
    val fahrenheit: Double = 98.6
    val celsius = fahrenheitToCelsius(fahrenheit)
    println(f"$fahrenheit%.2f°F is equal to $celsius%.2f°C")
  }

}
